src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,355 @@
+        if (baseUrl.contains("uat.hazelcast.cloud"))
+            return String.format("ua-%s", clusterId);
+        else if (baseUrl.contains("cloud.hazelcast.com"))
+            return String.format("pr-%s", clusterId);
+        else {
+            throw new CloudException("Given base url isn't known and connection name cannot be created");
+        }
+    }
+
+    // It replaces the  value of token with *** in a response body. Such as bearer token, discovery token.
+    private String maskValueOfToken(String json)
+    {
+        return json.replaceAll("\"token\":\"(.*?)\"", "\"token\":\"******\"");
+    }
+
+    // It creates a folder with name clusterId under /home/user/
+    // Then it downloads the certificates for the cluster and put them in the created folder
+    private String downloadCertificatesAndGetPath(String clusterId) throws CloudException {
+        try
+        {
+            String userHome = System.getProperty("user.home");
+            Path pathClusterId = Paths.get(userHome, clusterId);
+
+            // If folder with clusterId is there then certificates are already downloaded
+            if(!Files.exists(pathClusterId))
+            {
+                createFolderAndDownloadCertificates(pathClusterId, clusterId);
+            }
+            return Paths.get(pathClusterId.toString(), "certificates").toString() + File.separator;
+        }
+        catch(Exception e)
+        {
+            throw new CloudException("Problem occurred during certificates download, error message is: " + e.getMessage());
+        }
+    }
+
+    private void createFolderAndDownloadCertificates(Path pathClusterId, String clusterId) throws IOException {
+        Path destination;
+        Files.createDirectories(pathClusterId);
+        Path pathResponseZip = Paths.get(pathClusterId.toString(), "certificates.zip");
+
+        destination = Paths.get(pathClusterId.toString(), "certificates");
+        Files.createDirectories(destination);
+
+        String cert_url = String.format("%s/cluster/%s/certificate", baseUrl, clusterId);
+        Request request = new Request.Builder()
+                .url(cert_url)
+                .get()
+                .header("Accept", "application/zip")
+                .header("Authorization", String.format("Bearer %s", bearerToken))
+                .build();
+
+        call = client.newCall(request);
+        Response response = call.execute();
+        try (FileOutputStream stream = new FileOutputStream(pathResponseZip.toString())) {
+            stream.write(response.body().bytes());
+        }
+        ZipFile zipFile = new ZipFile(pathResponseZip.toString());
+        zipFile.extractAll(destination.toString());
+        new File(pathResponseZip.toString()).delete();
+    }
+
+    private List<String> checkEnvironmentVariables()
+    {
+        ArrayList<String> envVariables = new ArrayList<>();
+        if(System.getenv("BASE_URL") == null)
```

Comment:
```suggestion
        if (StringUtil.isNullOrEmpty(System.getenv("BASE_URL")))
```
similar for others as well.

Reply:
Somehow, it throws class not found exception. Adding it as a TODO maybe in the future, it will be refactored.